tests/test_packages/test_skills/test_tac_participation/test_behaviours.py

```diff
@@ -0,0 +1,327 @@
+
+import logging
+from pathlib import Path
+from typing import cast
+from unittest.mock import patch
+
+import pytest
+
+from aea.helpers.transaction.base import Terms
+from aea.protocols.dialogue.base import DialogueMessage
+from aea.test_tools.test_skill import BaseSkillTestCase, COUNTERPARTY_ADDRESS
+
+from packages.fetchai.protocols.oef_search.message import OefSearchMessage
+from packages.fetchai.protocols.tac.message import TacMessage
+from packages.fetchai.skills.tac_participation.behaviours import (
+    TacSearchBehaviour,
+    TransactionProcessBehaviour,
+)
+from packages.fetchai.skills.tac_participation.dialogues import TacDialogues
+from packages.fetchai.skills.tac_participation.game import Game, Phase
+
+from tests.conftest import ROOT_DIR
+
+
+class TestTacSearchBehaviour(BaseSkillTestCase):
+    """Test tac behaviour of tac participation."""
+
+    path_to_skill = Path(ROOT_DIR, "packages", "fetchai", "skills", "tac_participation")
+
+    @classmethod
+    def setup(cls):
+        """Setup the test class."""
+        super().setup()
+        cls.tac_search_behaviour = cast(
+            TacSearchBehaviour, cls._skill.skill_context.behaviours.tac_search
+        )
+        cls.game = cast(Game, cls._skill.skill_context.game)
+        cls.logger = cls.tac_search_behaviour.context.logger
+
+    def test_setup(self):
+        """Test the setup method of the tac_search behaviour."""
+        assert self.tac_search_behaviour.setup() is None
+        self.assert_quantity_in_outbox(0)
+
+    def test_act_i(self):
+        """Test the act method of the tac_search behaviour where phase is not PRE_GAME."""
+        # setup
+        self.game._phase = Phase.GAME
+
+        # operation
+        with patch.object(self.logger, "log"):
+            self.tac_search_behaviour.act()
+
+        # after
+        assert self.game.phase == Phase.GAME
+        self.assert_quantity_in_outbox(0)
+
+    def test_act_ii(self):
+        """Test the act method of the tac_search behaviour where phase is PRE_GAME."""
+        # setup
+        self.game._phase = Phase.PRE_GAME
+        mocked_query = "some_query"
+
+        # operation
+        with patch.object(self.game, "get_game_query", return_value=mocked_query):
+            with patch.object(self.logger, "log") as mock_logger:
+                self.tac_search_behaviour.act()
+
+        # after
+        self.assert_quantity_in_outbox(1)
+
+        # _search_for_tac
+        message = self.get_message_from_outbox()
+        has_attributes, error_str = self.message_has_attributes(
+            actual_message=message,
+            message_type=OefSearchMessage,
+            performative=OefSearchMessage.Performative.SEARCH_SERVICES,
+            to=self.skill.skill_context.search_service_address,
+            sender=self.skill.skill_context.agent_address
+            + "_"
+            + str(self.skill.skill_context.skill_id),
```

Comment:
we need to fix this

Reply:
How? Do you mean changing it to the usual `self.skill.skill_context.agent_address`, without the 
```
+ "_"
+ str(self.skill.skill_context.skill_id)
```
part?